src/shared/applyTemplate.js

```diff
@@ -0,0 +1,24 @@
+import { isUndefined, isFunction } from './typeof'
+
+export default function applyTemplate({ component, metaTemplateKeyName, contentKeyName }, headObject, template, chunk) {
+  if (isUndefined(template)) {
+    template = headObject[metaTemplateKeyName]
+    delete headObject[metaTemplateKeyName]
+  }
+
+  // return early if no template defined
+  if (!template) {
+    return false
+  }
+
+  if (isUndefined(chunk)) {
+    chunk = headObject[contentKeyName]
+  }
+
+  if (isFunction(template)) {
```

Comment:
I'd suggest a ternary here? :relaxed:

Reply:
This feels like a deja-vu, like if you have said this before to me ;)